hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -81,15 +82,18 @@ def from_dtype(
     allow_subnormal: Optional[bool] = None,
     exclude_min: Optional[bool] = None,
     exclude_max: Optional[bool] = None,
+    min_magnitude: Real = 0,
+    max_magnitude: Optional[Real] = None,
```

Comment:
```suggestion
    min_magnitude: Union[int, float] = 0,
    max_magnitude: Optional[Union[int, float]] = None,
```

We'd want to use the builtins as type hints as we don't want to leave ambiguity for non-builtin objects that instance check with the `numbers` abcs e.g. `isinstance(np.int64(1), numbers.Real)`.

Reply:
Nah, this is OK - see

https://github.com/HypothesisWorks/hypothesis/blob/6db5fc587536eb578dbd0c56b79295883fa5b79e/hypothesis-python/src/hypothesis/strategies/_internal/numbers.py#L49-L50

Reply:
Ah yes, that's what I had first. But then, I figured I might have a look at what `float()` is accepting, since it is passed to that anyways. Turns out, this [is used in `complex_numbers()`](https://github.com/felixdivo/hypothesis/blob/7d7411e44b33d03b7d8f6a5f5f71e0baf1cb18b5/hypothesis-python/src/hypothesis/strategies/_internal/core.py#L1628-L1629) (*not* `numbers.Real`):

https://github.com/felixdivo/hypothesis/blob/7d7411e44b33d03b7d8f6a5f5f71e0baf1cb18b5/hypothesis-python/src/hypothesis/strategies/_internal/core.py#L98-L103

Here is a explanation on how it is defined:
https://github.com/felixdivo/hypothesis/blob/7d7411e44b33d03b7d8f6a5f5f71e0baf1cb18b5/hypothesis-python/src/hypothesis/strategies/_internal/numbers.py#L49-L50

Reply:
Yep, posted at the same time. ^^